Script for benchmarking inference steps

GMNGeoffrey opened the pull request:
This focuses on the step level because it allows us to get results much faster and evaluate inference time separately from initialization overhead. The idea is that this is likely actually a better proxy for performance when creating many designs than running inference to design completion. It's also about 50 times faster, which is nice for iterative development.

I cleaned up a couple things in the regular inference run as well as I was adapting it.

GMNGeoffrey committed:
Script for benchmarking inference steps

This focuses on the step level because it allows us to get results much
faster and evaluate inference time separately from initialization
overhead. The idea is that this is likely actually a better proxy for
performance when creating many designs than running inference to design
completion. It's also about 50 times faster, which is nice for iterative
development.

I cleaned up a couple things in the regular inference run as well as I
was adapting it.

GMNGeoffrey committed:
Default to 1 design

tpopp commented on rfdiffusion/inference/benchmark_step.py:
```diff
@@ -0,0 +1,94 @@
+        device_name = torch.cuda.get_device_name(torch.cuda.current_device())
+        log.info(
+            f"Found GPU with device_name {device_name}. Will run RFdiffusion on {device_name}"
+        )
```
This logic is in a few places now right? Might be worth extracting it into a function.

GMNGeoffrey replied:
Done

GMNGeoffrey committed:
Merge branch 'brium' into benchmark-step

GMNGeoffrey committed:
Factor out GPU finding

Also moved option to require GPU to the inference config, leaving the
default as False there and True for benchmarking. This is a useful
option for inference also.

GMNGeoffrey committed:
Correctly calculate sampler steps

Technically you could use a different end step although we deliberately
don't do that for benchmarking because it actually changes the output of
sample step and we want to simulate the middle of a run. Also because
calculating the correct end step to put in order to achieve the desired
number of warmup and benchmarking steps seems convoluted.

GMNGeoffrey committed:
Output time after each design
